lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -716,6 +720,21 @@ def _get_node_information(self, node: Node) -> Dict[str, str]:
             node.log.debug("detecting vm generation...")
             information[KEY_VM_GENERATION] = node.tools[VmGeneration].get_generation()
             node.log.debug(f"vm generation: {information[KEY_VM_GENERATION]}")
+
+            if node.capability and node.capability.features:
+                for feat in node.capability.features:
```

Comment:
Why does it check the capability of node, not check by node.features.is_supported? The capability is what's supported by the node, but it's not the runtime value.

Reply:
@squirrelsc 
Since this logic only runs when `node.is_connected` the security profile capability should be resolved to the runtime value. I used the some of the logic from `SecurityProfile.on_before_deployment`.

Right now, `SecurityProfile.enabled` returns True and `SecurityProfie.create_settings` always returns a value. But I think your point is in cases where SecurityProfile is disabled or no settings are returned?

I can try to make this logic more robust, but **can you help me understand the scenario in which you expect it will break?**

Reply:
@squirrelsc I believe I have fixed this issue. Please see latest changes :)